### PR TITLE
(fix): wdio plugin issue with wdio v6

### LINKED ISSUE #2315: Error: Could not load plugin wdio from module './plugin/wdio': Cannot read property 'outputDir' of undefined
I get this error when I try to run the case using command:
npx codeceptjs run --steps

Error:
Could not load plugin wdio from module './plugin/wdio':
Cannot read property 'outputDir' of undefined

Here is the package.json:
{
  "name": "example",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@wdio/cli": "^6.0.13",
    "@wdio/selenium-standalone-service": "^6.0.12",
    "codeceptjs": "^2.6.0",
    "webdriver": "^6.0.13"
  },
  "dependencies": {
    "webdriverio": "^6.0.13"
  }
}
Looking for some solution at the earliest. Thanks.

### PR DESCRIPTION
**Pull request by PeterNgTr**
## Motivation/Description of the PR
- Due to the changes on the way how services is initialized in wdio, we need to update our wdio plugin to make it work again with wdio v6.
- Resolves #2315, #2288.

Applicable helpers:

- [x] WebDriver

Applicable plugins:

- [x] wdio

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [x] :bug: Bug fix
- [ ] :clipboard: Documentation changes/updates
- [x] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ lib/helper/WebDriver.js  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=5311638&bid=17381947)

**Comment by pablopaul** [comment]
@PeterNgTr can you add a test?

**Comment by PeterNgTr** [comment]
@pablopaul I'm not sure if it's easy to set up the test evn for this :|

**Comment by DavertMik** [comment]
I agree, we can't set up tests. Esp because you can choose any wdio service with this...

**Comment by pablopaul** [comment]
@DavertMik if we can choose any we can also pick exactly one wido service for test :)

And I am fine not adding a test here if it is too complex 👍 approved!

File: lib/plugin/wdio.js [hunk 1]
```diff
@@ -99,7 +99,8 @@ module.exports = (config) => {
     if (Service) {
       if (Service.launcher && typeof Service.launcher === 'function') {
         const Launcher = Service.launcher;
-        launchers.push(new Launcher(config));
+        const options = { logPath: global.output_dir, installArgs: {}, args: {} };
+        launchers.push(new Launcher(options, [config.capabilities], config));
```
Comment (DavertMik) [inline lib/plugin/wdio.js:103]: is this backwards compatible with webdriverio 5?

Comment (PeterNgTr) [reply lib/plugin/wdio.js:103]: With that code, no. 😅. Maybe need to add the condition to make it backwards compatible with v5.

**Comment by vrao4160** [comment]
Hi,

Is this output dir issue resolved ?

How to get fix for this soon?

File: lib/plugin/wdio.js (hunk 1 above)
Comment (PeterNgTr) [reply lib/plugin/wdio.js]: @DavertMik @pablopaul made it be compatible with v5.

**Comment by codejamninja** [comment]
Is this fixed?

**Comment by x86chi** [comment]
It not working in 2.6.3 :/

```bash
$ /node_modules/.bin/codeceptjs run
DEPRECATION NOTICE:
You are using webdriverio v5. It is recommended to update to webdriverio@6.
Support of webdriverio v5 is deprecated and will be removed in CodeceptJS 3.0


Could not load plugin wdio from module './plugin/wdio':
Cannot read property 'outputDir' of undefined
```

**Comment by diliyaer** [comment]
> It not working in v2.6.3 :/
> 
> ```shell
> $ /node_modules/.bin/codeceptjs run
> DEPRECATION NOTICE:
> You are using webdriverio v5. It is recommended to update to webdriverio@6.
> Support of webdriverio v5 is deprecated and will be removed in CodeceptJS 3.0
> 
> 
> Could not load plugin wdio from module './plugin/wdio':
> Cannot read property 'outputDir' of undefined
> ```

if I guessed right your package.json only has 'webdriver' but not 'webdriverio??

here is the workaround:
1. run cmd "npm install webdriverio" on your project directory to install latest webdriverio
2. it should add below dependencies in package.json:
 "dependencies": {
    "webdriverio": "^6.1.9"
  }
3. codecept.conf.js should have:
plugins: {
    ... ,
    wdio: {
      enabled: true,
      services: ['selenium-standalone']
    }
  }

mine is working now, hope it helps.

**Comment by sdeliyar** [comment]
> It not working in v2.6.3 :/
> 
> ```shell
> $ /node_modules/.bin/codeceptjs run
> DEPRECATION NOTICE:
> You are using webdriverio v5. It is recommended to update to webdriverio@6.
> Support of webdriverio v5 is deprecated and will be removed in CodeceptJS 3.0
> 
> 
> Could not load plugin wdio from module './plugin/wdio':
> Cannot read property 'outputDir' of undefined
> ```

if I guessed right your package.json only has 'webdriver' but not 'webdriverio??

here is the workaround:

run cmd "npm install webdriverio" on your project directory to install latest webdriverio
it should add below dependencies in package.json:
"dependencies": {
"webdriverio": "^6.1.9"
}
codecept.conf.js should have:
plugins: {
... ,
wdio: {
enabled: true,
services: ['selenium-standalone']
}
}
mine is working now, hope it helps.

**Comment by x86chi** [comment]
thanks. it work!

**Comment by vaz0711** [comment]
Unfortunately, It doesn't help, if webdriverio has installed as a global package :(

**Comment by rightwinger** [comment]
$ npm install webdriverio  
It worked for me.
You need t check that  webdriver, codeceptjs, chrome are properly defined or are latest.

**Comment by lind-eduard** [comment]
Checked everything twice, still same error. I have the latest version of everything, reinstalled it 3 times

**Comment by R-terraDev** [comment]
> Checked everything twice, still same error. I have the latest version of everything, reinstalled it 3 times
@lind-eduard 
@DavertMik 
I have a very recently initialized codeceptjs project and im seeing this problem as well.

After I removed the @codeceptjs/configure package from my deps the issue resolved itself

**Comment by lind-eduard** [comment]
@R-terraDev Thanks for hint, it worked for me....almost. After that I got an error "unknown command: session/e60c2044d3d5204bea899837f9c5b569/se/log", I already saw this when I had a difference in local version of driver and selenium version, but when I am trying to set version for wdio plugin I get another error: "Could not load plugin wdio from module './plugin/wdio':
name.toLowerCase is not a function"

**Comment by khanzzirfan** [comment]
// Not working

"devDependencies": {
    "@codeceptjs/configure": "^0.5.2",
    "@reportportal/agent-js-codecept": "^0.2.3",
    "@wdio/cli": "^6.3.6",
    "@wdio/selenium-standalone-service": "^6.1.14",
    "allure-commandline": "^2.13.0",
    "codeceptjs": "^2.6.6",
    "webdriverio": "^6.1.23"
  }

// working 
"devDependencies": {
    "@reportportal/agent-js-codecept": "^0.2.3",
    "@wdio/cli": "^6.3.6",
    "@wdio/selenium-standalone-service": "^6.1.14",
    "allure-commandline": "^2.13.0",
    "codeceptjs": "^2.6.6",
    "webdriverio": "^6.1.23"
  }

don't know what is wrong with @codeceptjs/configure. may be its referering to old wdio library.

**Pull request merged**